### PR TITLE
Registration of rule

### LINKED ISSUE #37: Register rule in TokenRules
- [x] Whitelisted organization worker registers rule
- [x] Command
	registerRule config.json ruleName ruleAddress ruleABI.
- [x] Readme update

### PR DESCRIPTION
**Pull request by abhayks1**
PR covers below:
- JLP Command to register rule to TokenRules.
```
node ./src/bin/register_rule.js <config.json> <ruleName> <ruleAddress> <ruleAbi>
```
- Readme update.

- Worker is added while deploying Organization contract.

Fixes #37

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,25 +209,42 @@ node src/bin/facilitator.js progressRedeem path_to_config.json messageHash
...
+  node ./src/bin/registerRule.js --help
+  
+  # Register rule to TokenRules:
+  node ./src/bin/registerRule.js config.json ruleName ruleAddress ruleAbi
+  ```
+  
+  * Replace `config.json` with the path to the configuration file.
+  * Replace `ruleName` with name of the rule.
+  * Replace `ruleAddress` with address of the rule. 
+  * Replace `ruleAbi` with abi of the rule.
```
Comment (schemar) [inline README.md]: Why did you add a leading space to every line?

File: src/config/chain_config.js [hunk 2]
```diff
@@ -13,11 +13,11 @@ class ChainConfig {
     this.auxiliaryChainId = process.env.ORIGIN_CHAIN_ID || 200;
     this.simpleTokenAddress = process.env.SIMPLE_TOKEN_ADDRESS || '0xca954C91BE676cBC4D5Ab5F624b37402E5f0d957';
     this.originMasterKey = process.env.ORIGIN_MASTER_KEY || '0x6d4e5f5ca54b88fdb68a5d0e6ea5aa3869f88116';
-    this.auxiliaryMasterKey = process.env.AUXILIARY_MASTER_KEY || '0x490cc731b076e58509e1f7cee6c68b39cfe5ad62';
+    this.auxiliaryMasterKey = process.env.AUXILIARY_MASTER_KEY || '0x12b38720245e7bf06428a433a34b90d77e0198e1';
     this.originBurnerAddress = process.env.ORIGIN_BURNER_ADDRESS || '0x0000000000000000000000000000000000000000';
     this.auxiliaryBurnerAddress = process.env.AUXILIARY_BURNER_ADDRESS || '0x0000000000000000000000000000000000000000';
     this.originDeployerAddress = process.env.ORIGIN_DEPLOYER_ADDRESS || '0x6d4e5f5ca54b88fdb68a5d0e6ea5aa3869f88116';
-    this.auxiliaryDeployerAddress = process.env.AUXILIARY_DEPLOYER_ADDRESS || '0x490cc731b076e58509e1f7cee6c68b39cfe5ad62';
+    this.auxiliaryDeployerAddress = process.env.AUXILIARY_DEPLOYER_ADDRESS || '0x12b38720245e7bf06428a433a34b90d77e0198e1';
```
Comment (schemar) [inline src/config/chain_config.js]: Why are you changing the defaults? You should not need to.

File: src/bin/registerRule.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+program
+  .version(version)
+  .name('RegisterRule')
+  .arguments('<config> <ruleName> <ruleAddress> <ruleAbi>')
+  .description('An executable to register rules in tokenrules.')
+  .action(
+    async (config, ruleName, ruleAddress, ruleAbi) => {
+      const chainConfig = new ChainConfig(config);
+      const registerRuleTxOptions = {
```
Comment (schemar) [inline src/bin/registerRule.js]: You should use `connected::run()` to assert correct signing of transactions and closing of the connection.

File: src/config/chain_config.js (hunk 2 above)
Comment (abhayks1) [reply src/config/chain_config.js]: Agree!. It's  by mistake 😞. Will update.

**Comment by abhayks1** [comment]
@schemar changes has been done. I will be testing openst JLP pieces together once with the openst setup JLP.

File: README.md [hunk 4]
```diff
@@ -213,25 +213,42 @@ node src/bin/facilitator.js progressRedeem path_to_config.json messageHash
...
-node ./src/bin/tokenrules.js config.json eip20Token organization
-```
-
-It will write `tokenRulesAddress` to your config file.
-
+  * `messageHash` generated in redeem facilitator step.
+  
+ ## Deploy TokenRules
+ 
+ Prerequisite: `eip20Token` and `organization` in your config file.
```
Comment (sarvesh-ost) [inline README.md]: on the auxiliary chain right? 
It's good to add this explicitly.

File: src/bin/registerRule.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Web3 = require('web3');
+const OpenST = require('@openstfoundation/openst.js');
+
+const connected = require('../connected');
+const ChainConfig = require('../config/chain_config');
```
Comment (sarvesh-ost) [inline src/bin/registerRule.js]: This is not used now,  I think.

File: src/bin/registerRule.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+          };
+          const auxiliaryWeb3 = new Web3(connection.auxiliaryWeb3);
+          const tokenRules = new OpenST.ContractInteract.TokenRules(
+            auxiliaryWeb3,
+            chainConfig.openst.tokenRules,
+          );
+          await tokenRules.registerRule(ruleName, ruleAddress, ruleAbi, registerRuleTxOptions);
+          logger.info(`Rule ${ruleName} registered!`);
+          logger.info('Validating registered rule...');
+          const rule = await tokenRules.getRuleByName(ruleName);
```
Comment (sarvesh-ost) [inline src/bin/registerRule.js]: Do we need to make another call to verify rule is registered?  
Receipt status should also tell if the rule is registered. It could save one extra geth call.

Comment (abhayks1) [reply src/bin/registerRule.js]: Do you see an optimization concern in JLP 😮 ?
The call also tests getByRuleName interact method. 
This has a little advantage of verifying with matching exact name and address 🤔  .

Comment (sarvesh-ost) [reply src/bin/registerRule.js]: Why not? It's extra code/call.
